install/kubernetes/ingress-sni-zvpn.yaml

```diff
@@ -0,0 +1,364 @@
+      #            config:
+      #              stat_prefix: http443
+      #              codec_type: AUTO
+      #              rds:
+      #                route_config_name: "443"
+      #                config_source:
+      #                  api_config_source:
+      #                      api_type: REST_LEGACY
+      #                      refresh_delay: 10s
+      #                      cluster_names: [rds]
+
+      # Example for testing explicit routes (route_config vs rds)
+      - name: http8081
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 8081 }
+        filter_chains:
+        - filters:
+          - name: envoy.http_connection_manager
+            # TODO: add mixer
+            config:
+              stat_prefix: ingress_http8080
+              codec_type: AUTO
+              route_config:
+                name: local_route
+                virtual_hosts:
+                - name: local_service
+                  domains: ["*"]
+                  routes:
+                  - match: { prefix: "/" }
+                    route: { cluster: out.pilot }
+
+
+      - name: tcp-zvpn-pilot
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 15003 }
+        filter_chains:
+        - filters:
+          - name: envoy.tcp_proxy
+            config:
+              # listener_manager_impl.cc - if false, will attempt to convert to proto - not implemented for tcp_proxy
+              deprecated_v1: true
+              value:
+                stat_prefix: tcp_pilot
+                route_config:
+                    routes:
+                    - cluster: zvpn.pilot
+
+      - name: tcp-zvpn-ca
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 8060 }
+        filter_chains:
+        - filters:
+          - name: envoy.tcp_proxy
+            config:
+              # listener_manager_impl.cc - if false, will attempt to convert to proto - not implemented for tcp_proxy
+              deprecated_v1: true
+              value:
+                stat_prefix: tcp_ca
+                route_config:
+                    routes:
+                    - cluster: zvpn.ca
+
+
+      clusters:
+      - name: rds
+        connect_timeout: 0.25s
+        type: STRICT_DNS
+        lb_policy: ROUND_ROBIN
+        hosts:
+          # TODO(costin): add TLS
+          - { socket_address: { address: "istio-pilot.istio-system", port_value: 8080 }}
+          # Local debugging/testing
+          #- { socket_address: { address: "localhost", port_value: 15008 }}
+
+      # Direct zvpn connection to pilot TLS port. This is used with a TCP proxy !
+      - name: zvpn.pilot
+        connect_timeout: 0.25s
+        type: STRICT_DNS
+        lb_policy: ROUND_ROBIN
+        hosts:
+          - { socket_address: { address: "istio-pilot", port_value: 15003 }}
+
+      # Connection to Istio CA. Service name must match the one for CA exposing gRPC.
+      - name: zvpn.ca
+        connect_timeout: 0.25s
```

Comment:
No TLS, no protocol options?

Reply:
No protocol options as it’s a tcp cluster. No tls either since it’s passthrough tcp cluster

Reply:
Yes, that's the point of zvpn - TLS is end to end, this is just TCP proxy. We may add some 
extra options ( customizable timeout, etc)